src/main/java/org/cactoos/map/Synced.java

```diff
@@ -139,16 +156,19 @@ public Synced(final Iterable<Map.Entry<X, Y>> list) {
      * @param map Pre-existing map we want to extend
      * @param list Entries for the entries
      */
-    public Synced(final Map<X, Y> map,
-        final Iterable<Map.Entry<X, Y>> list) {
+    public Synced(
+        final Map<? extends X, ? extends Y> map,
+        final Iterable<Map.Entry<? extends X, ? extends Y>> list
+    ) {
         this(new MapOf<>(map, list));
     }
 
     /**
      * Ctor.
      * @param map The map
      */
-    public Synced(final Map<X, Y> map) {
-        super(Collections.synchronizedMap(map));
+    @SuppressWarnings("unchecked")
+    public Synced(final Map<? extends X, ? extends Y> map) {
+        super(Collections.synchronizedMap(new MapOf<>(map)));
```

Comment:
@andreoss here we cannot use `MapOf` because it would create a new map instead of making the wrapped one synchronized.

Reply:
@victornoel Done.